message_endpoint.go

```diff
@@ -0,0 +1,29 @@
+package swim
+
+// I think this Message struct should come from pb?
+type Message struct{}
+
+// MessageEndpoint basically do receiving packet and determine
+// which logic executed based on the packet.
+type MessageEndpoint struct {
+	transport UDPTransport
+}
+
+// ListenPacket is a log running goroutine that pulls packet from the
+// transport and pass it for processing
+func (m *MessageEndpoint) ListenPacket() {
+
```

Comment:
I think just `Listen` or `Start` is fine, I want to hide `Packet` because it is a low abstraction level.

Reply:
`Listen` would be good!